lib/utils.js

```diff
@@ -10,23 +9,22 @@ var isBuffer = require('is-buffer');
 var toString = Object.prototype.toString;
 
 /**
- * Determine if a value is an Array
+ * Determine if a value is an ArrayBuffer
  *
  * @param {Object} val The value to test
- * @returns {boolean} True if value is an Array, otherwise false
+ * @returns {boolean} True if value is an ArrayBuffer, otherwise false
  */
-function isArray(val) {
-  return toString.call(val) === '[object Array]';
+function isArrayBuffer(val) {
+  return toString.call(val) === '[object ArrayBuffer]';
 }
 
 /**
- * Determine if a value is an ArrayBuffer
- *
+ * Determine if a value is a Buffer
  * @param {Object} val The value to test
- * @returns {boolean} True if value is an ArrayBuffer, otherwise false
+ * @returns {boolean} true if value is a Buffer, otherwise false
  */
-function isArrayBuffer(val) {
-  return toString.call(val) === '[object ArrayBuffer]';
+function isBuffer(val) {
+  return val && val.constructor && typeof val.constructor.isBuffer === 'function' && val.constructor.isBuffer(val);
```

Comment:
The nice thing about the dependency was that it had tests, do you mind adding some tests?